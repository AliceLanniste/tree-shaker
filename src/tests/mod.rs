fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
