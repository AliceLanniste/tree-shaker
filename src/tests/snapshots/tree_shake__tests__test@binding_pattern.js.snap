---
source: src/tests/mod.rs
expression: tree_shake(&input)
input_file: src/tests/fixtures/binding_pattern.js
---
export function simple(__unused_55E7) {
	effect(1);
	effect(2);
	effect(3);
	effect(4);
	let { g: { h: __unused_83AC, i: { j: __unused_C4E6 } } } = unknown;
}
export function with_rest(unknown) {
	let { a: __unused_AA1B,...rest } = {
		a: 1,
		b: unknown,
		c: 2
	};
	effect(1, rest.b);
	let { b: __unused_19B7,...__unused_DEBA } = unknown;
	let { c: __unused_51E4,...__unused_BE17 } = rest2;
}
